application/src/main/webapp/vue-app/components/ExoChatContactList.vue

```diff
@@ -403,28 +403,30 @@ export default {
       const message = event.detail;
       const room = message.room;
 
-      if(!room) {
+      if (!room) {
         return;
       }
 
       const foundContact = this.findContactByRoomOrUser(room, message.data ? message.data.user : message.sender);
 
-      if(foundContact) {
+      if (foundContact) {
         if (!foundContact.lastMessage) {
           foundContact.lastMessage = {};
         }
         foundContact.lastMessage = message.data;
         foundContact.timestamp = message.ts;
-        if (this.selected){
+        if (this.selected) {
           if (this.selected.room !== foundContact.room || this.mq === 'mobile') {
-            foundContact.unreadTotal ++;
+            chatServices.getNotReadMessagesByRoom(eXo.chat.userSettings, chatConstants.CHAT_NOTIFICATION_TYPE, chatConstants.CHAT_NOTIFICATION_CATEGORY, foundContact.room, true).then((data) => {
```

Comment:
There is no need to send a request to the server to increment the number of total messages. We know the number of unread messages and we need just to incrementit.

Reply:
The event of receiving message is invoked more than once each time the user receive a message, so the number of messages will be incremented two times, thus i choose to get the exact value from the server